AlpsSDK/ProximityHandler.swift

```diff
@@ -0,0 +1,233 @@
+//
+//  ProximityHandler.swift
+//  AlpsSDK
+//
+//  Created by Wen on 26.10.17.
+//  Copyright © 2017 Alps. All rights reserved.
+//
+
+import Foundation
+import CoreLocation
+import Alps
+
+class ProximityHandler {
+    
+    var contextManager: ContextManager!
+    static var immediateTrigger: [String: ProximityEvent] = [:]
+    static var nearTrigger: [String: ProximityEvent] = [:]
+    static var farTrigger: [String: ProximityEvent] = [:]
+    static var unknownTrigger: [String: ProximityEvent] = [:]
+    var immediateBeacons: [String] = []
+    var nearBeacons: [String] = []
+    var farBeacons: [String] = []
+    var unknownBeacons: [String] = []
+    var refreshTimer: Int = 60 * 1000 // timer is in milliseconds
+    
+    init(contextManager: ContextManager) {
+        self.contextManager = contextManager
+    }
+    
+    // Parsing beacons proximity relatively to the device that detect them
+    func parseBeaconsByProximity(_ beacons: [CLBeacon]) {
+        var ourBeacon: IBeaconTriple?
+        for beacon in beacons {
+            let b = syncBeacon(beacon: beacon)
+            if !b.isEmpty {
+                ourBeacon = b[0]
+            }
+            guard let deviceId = ourBeacon?.deviceId else {
+                return
+            }
+            removeDuplicate(clBeacon: beacon, deviceId: deviceId)
+        }
+    }
+    
+    // Launch a proximity event
+    func triggerBeaconsProximityEvent() {
+        for clProximity in CLProximity.allValues {
+            var (beacons, trigger, distance) = setUpTriggerBeaconsProximityEvent(forCLProximity: clProximity)
+            for id in beacons {
+                // Check if a proximity event already exist
+                if trigger[id] == nil {
+                    // Send the proximity event
+                    let proximityEvent = ProximityEvent.init(deviceId: id, distance: distance)
+                    let deviceId = "NEED CHANGE to a guard"
+                        sendProximityEvent(deviceId: deviceId, proximityEvent: proximityEvent) {(_ proximityEvent) in
```

Comment:
we should be able to send batch of proximity evens. Now it will trigger request for very beacon. We could aggregate them.

@adamsmo what do you think?